external-crates/move/crates/move-compiler/src/typing/translate.rs

```diff
@@ -2204,12 +2204,54 @@ fn match_pattern_(
             }
             let bt = rtype!(bt);
             let pat_ = if let Some(mut_) = mut_ref {
-                TP::BorrowConstructor(*mut_, m, enum_, variant, targs, tfields)
+                TP::BorrowVariant(*mut_, m, enum_, variant, targs, tfields)
             } else {
-                TP::Constructor(m, enum_, variant, targs, tfields)
+                TP::Variant(m, enum_, variant, targs, tfields)
             };
             T::pat(bt, sp(loc, pat_))
         }
+        P::Struct(m, struct_, tys_opt, fields) => {
+            let (bt, targs) = core::make_struct_type(context, loc, &m, &struct_, tys_opt);
+            let typed_fields = add_struct_field_types(
+                context,
+                loc,
+                "pattern",
+                &m,
+                &struct_,
+                targs.clone(),
+                fields,
+            );
+            let tfields = typed_fields.map(|f, (idx, (fty, tpat))| {
+                let tpat = match_pattern_(context, tpat, mut_ref, rhs_binders, wildcard_needs_drop);
+                let fty_ref = rtype!(fty.clone());
+                subtype(
+                    context,
+                    f.loc(),
+                    || "Invalid pattern field type",
+                    tpat.ty.clone(),
+                    fty_ref,
+                );
+                (idx, (fty, tpat))
+            });
+            if !context.is_current_module(&m) {
+                let msg = format!(
+                    "Invalid deconstructing pattern for '{}::{}'.\n All struct can only be \
```

Comment:
Nit on wording here. Consider "Invalid pattern for '{}::{}'.\nStructs can only be \"

Reply:
Changed both places this occurred